src/main.rs

```diff
@@ -47,34 +47,37 @@ fn get_registry() -> Vec<(Principal, Vec<u64>)> {
 }
 
 #[query]
-fn tokens(user: Principal) -> Vec<u64> {
+fn soulbound_tokens(user: Principal) -> Vec<u64> {
     State::read_state(|state| state.get_user_registry(user))
 }
 
 #[update(guard = "is_admin")]
-fn add_asset(asset: Vec<u8>) -> Result<(), String> {
+fn add_asset(asset_name: String, asset: Vec<u8>) -> Result<(), String> {
     State::mutate_state(|state| {
```

Comment:
I wonder if it's worth checking memory usage here. What happens if we exceed canister memory?